libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -278,7 +280,10 @@ internal class MapboxTripSession(
 
             val remainingWaypoints =
                 ifNonNull(tripStatus.route?.routeOptions()?.coordinatesList()?.size) {
-                    it - tripStatus.navigationStatus.nextWaypointIndex
+                    it - max(
+                        tripStatus.navigationStatus.nextWaypointIndex,
+                        FALLBACK_NEXT_WAYPOINT_INDEX_IF_NATIVE_NAVIGATOR_MAKES_MISTAKE
+                    )
```

Comment:
@averkhaturau , did I understand the suggested workaround correctly? If `nextWaypointIndex ` is 0 SDK should use 1.

Reply:
Personally I don't like a name of the constant, logically it's just `max(1, tripStatus.navigationStatus.nextWaypointIndex)`. In navigator logic 0 is correct just after route update when it's not clear if we are on route.

Reply:
@averkhaturau ,
> In navigator logic 0 is correct just after route update when it's not clear if we are on route.

What would be the `routeState` then? `INITIALIZED`? 

which states except `INVALID` could return 0 and how we're supposed to handle it? As far as I understand if we get 0 with `OFF_ROUTE` state it's safe to fallback to 1. What are the other cases?

Reply:
Right now SDK converts the data it gets in status update and pass it to the user unless state is `RouteState.INVALID`. I'm wondering is that correct behaviour?

Reply:
@VysotskiVadim, Please do not rely explicitly on RouteState, it's determined differently. It means, we do not define how RouteState and nextWaypointIndex correlate in the status. I can definitely say that in case of nextWaypointIndex==0 it's not `Tracking`, but from the description it can be Invalid or Initialized.

Reply:
@averkhaturau , if the Android SDK shouldn't rely on the status, how can the SDK understand if the value in `nextWaypointIndex` can be trusted?
Trust `nextWaypointIndex` if it's not 0, right?

Reply:
> Trust nextWaypointIndex if it's not 0, right?

True for last route set into navigator. 0 can be also trusted, but it may mean 'no route' as well as 'not yet on route'